src/OpenLoco/Map/TileManager.cpp

```diff
@@ -188,6 +210,92 @@ namespace OpenLoco::Map::TileManager
         return std::make_tuple(height, waterHeight);
     }
 
+    std::tuple<int16_t, int16_t> getHeight(map_pos pos)
+    {
+        return getHeight(pos.x, pos.y);
+    }
+
+    int16_t getLandOrWaterHeight(map_pos pos)
+    {
+        auto [land, water] = TileManager::getHeight(pos);
+        return water == 0 ? land : water;
+    }
+
+    static void clearTilePointers()
+    {
+        tile_element** tiles = _tiles;
+        std::memset(tiles, 0xFF, map_pitch * map_columns * sizeof(tile_element*));
+    }
+
+    static void set(TilePos pos, tile_element* elements)
+    {
+        _tiles[(pos.y * map_pitch) + pos.x] = elements;
+    }
+
+    // 0x00461348
+    void updateTilePointers()
+    {
+        clearTilePointers();
+
+        tile_element* el = _elements;
+        for (tile_coord_t y = 0; y < map_rows; y++)
+        {
+            for (tile_coord_t x = 0; x < map_columns; x++)
+            {
+                set(TilePos(x, y), el);
+
+                // Skip remaining elements on this tile
+                do
+                {
+                    el++;
+                } while (!(el - 1)->isLast());
+            }
+        }
+
+        _elementsEnd = el;
+    }
+
+    // 0x0046148F
+    void reorganise()
+    {
+        Ui::setCursor(Ui::cursor_id::busy);
+
+        auto tempBuffer = reinterpret_cast<tile_element*>(std::malloc(maxElements * sizeof(tile_element)));
+        if (tempBuffer == nullptr)
+        {
+            exitWithError(4370, StringIds::null);
+            return;
+        }
+
+        size_t numElements = 0;
+        for (tile_coord_t y = 0; y < map_rows; y++)
+        {
+            for (tile_coord_t x = 0; x < map_columns; x++)
+            {
+                auto tile = get(TilePos(x, y));
+                for (auto element : tile)
+                {
+                    tempBuffer[numElements] = element;
+                    numElements++;
+                }
+            }
+        }
+
+        // Copy organised elements back to original element buffer
+        std::memcpy(_elements, tempBuffer, numElements * sizeof(tile_element));
+
+        // Zero all unused elements
+        auto remainingElements = maxElements - numElements;
+        std::memset(_elements + numElements, 0, remainingElements * sizeof(tile_element));
+
+        std::free(tempBuffer);
```

Comment:
Pretty sure we can use a smart pointer rather than having to do this or even a vector.